File: app/src/main/java/io/github/ovso/righttoknow/framework/network/api/NewsApi.java
package io.github.ovso.righttoknow.framework.network.api;

import io.github.ovso.righttoknow.news.model.NewsResult;
import io.reactivex.Single;
import java.util.HashMap;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.QueryMap;

/**
 * Created by jaeho on 2017. 12. 20
 */

public interface NewsApi {

  @GET("/v1/search/news") Single<NewsResult> getNews(@QueryMap HashMap<String, String> queryMap);

  @GET("/v1/search/news") Single<NewsResult> getNews(@Query("query") String query,
      @Query("display") int display, @Query("start") int start, @Query("sort") String sort);
}

File: app/src/main/java/io/github/ovso/righttoknow/framework/network/api/NewsApi.kt
package io.github.ovso.righttoknow.framework.network.api

import io.github.ovso.righttoknow.ui.main.news.model.NewsResult
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.QueryMap
import java.util.*

interface NewsApi {
  @GET("/v1/search/news")
  fun getNews(@QueryMap queryMap: HashMap<String, String>): Single<NewsResult>

  @GET("/v1/search/news")
  fun getNews(
    @Query("query") query: String,
    @Query("display") display: Int,
    @Query("start") start: Int,
    @Query("sort") sort: String): Single<NewsResult>
}